updater.js

```diff
@@ -0,0 +1,233 @@
+      versionNumberRegexp.lastIndex = 0;
+      let regexpResult = versionNumberRegexp.exec(versionString);
+      if (regexpResult) {
+        return new Version(regexpResult);
+      }
+    }
+
+    throw new Error("Can't parse version string : syntax error");
+  }
+
+  toString() {
+    let string = this.prefix + this.major + "." + this.minor;
+    string += this.patch !== null ? "." + this.patch : "";
+    string += this.fourth !== null ? "." + this.fourth : "";
+    return string;
+  }
+}
+
+class Update {
+  constructor(walletIdentifier, from, to) {
+    this.walletIdentifier = walletIdentifier;
+    this.from = from;
+    this.to = to;
+  }
+
+  toString() {
+    return "updated " + this.walletIdentifier + " from " + this.from.toString() + " to " + this.to.toString();
+  }
+}
+
+(async function() {
+  try {
+    console.log("--- Wallet updater launched at " + new Date() + "--");
+    let updates = await checkAllForUpdates();
+
+    if (updates.filter(o => { return o !== null; }).length > 0) {
+      await updateFile(updates);
+      await updatePatchNumber();
+
+      const commitMessage = buildCommitMessage(updates);
+
+      await addDeployKey();
+      await addChanges();
+      await commit(commitMessage);
+      await tag(commitMessage);
+      await push();
+      await publish();
+    } else {
+      console.log("Wallets are up to date.");
+    }
+
+    console.log("--- Wallet updater ended at " + new Date() + "--");
+  } catch (e) {
+    console.error(e);
+  }
+})();
+
+
+async function addDeployKey() {
+  if (!process.env.deploy_key) {
+    throw new Error("Environment variable deploy_key is not set - cannot send modifications to server.");
+  }
+  const { stdout, stderr } = await exec("eval \"$(ssh-agent -s)\" && echo $deploy_key | ssh-add -");
+  console.log(stdout);
+  console.log(stderr);
+}
+
+async function addChanges() {
+  const { stdout } = await exec("git add wallets.json package.json");
+  console.log(stdout);
+}
+
+async function commit(message) {
+  const { stdout } = await exec("git commit -m \"" + message + "\"");
+  console.log(stdout);
+}
+
+async function tag(message) {
+  const { stdout } = await exec("git tag " + manifest.version + " -m \" " + message + "\"");
+  console.log(stdout);
+}
+
+async function push() {
+  const { stderr } = await exec("git push --follow-tags origin HEAD");
+  console.log(stderr);
+  console.log("Updated wallet.json successfully");
+}
+
+function buildCommitMessage(updates) {
+  let commitMessage = "Update wallet.json\n\n";
+  updates.forEach(update => {
+    if (update) {
+      commitMessage += update.toString() + "\n";
+    }
+  });
+  return commitMessage;
+}
+
+async function updateFile(updates) {
+  updates.forEach(update => {
+    if (update) {
+      wallets[update.walletIdentifier].tag = update.to.toString();
+    }
+  });
+  return writeFile("./wallets.json", JSON.stringify(wallets, null, "  "));
+}
+
+async function checkAllForUpdates() {
+  const pendingUpdates = [];
```

Comment:
Seems convoluted, even If I know we use to do things this way....
Might get something better with something like:
```
const pendingUpdates = Object.keys(wallets).map(key => checkForUpdates(key, wallets[key]))
```.